testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MsalWrapper.java

```diff
@@ -148,14 +152,33 @@ public void acquireToken(final Activity activity,
             return;
         }
 
-        AcquireTokenParameters parameters = new AcquireTokenParameters.Builder()
-                .startAuthorizationFromActivity(activity)
+        final AcquireTokenParameters.Builder builder = new AcquireTokenParameters.Builder();
+        builder.startAuthorizationFromActivity(activity)
                 .withScopes(Arrays.asList(requestOptions.getScopes().toLowerCase().split(" ")))
                 .withLoginHint(requestOptions.getLoginHint())
                 .withPrompt(requestOptions.getPrompt())
-                .withOtherScopesToAuthorize(Arrays.asList(requestOptions.getExtraScopesToConsent().toLowerCase().split(" ")))
-                .withCallback(getAuthenticationCallback(notifyCallback))
-                .build();
+                .withOtherScopesToAuthorize(
+                        Arrays.asList(
+                                requestOptions
+                                        .getExtraScopesToConsent()
+                                        .toLowerCase()
+                                        .split(" ")
+                        )
+                )
+                .withCallback(getAuthenticationCallback(notifyCallback));
+
+        if (USE_POP) {
+            try {
+                builder.withProofOfPossession(
+                        new URL("https://signedhttprequest.azurewebsites.net/api/validateSHR"),
```

Comment:
Should either remove this or add new controls to testapp to support PoP testing....